### PR TITLE
Boat Page - Layout & Design

### PR DESCRIPTION
**Pull request by mienaikoe**
- Drag and Drop for Boat Page
- Simplification and Breakout of Boat Page components
- Spacing & Styling

### CODE REVIEW

File: client/pages/Boat/SeatPicker.js [hunk 1]
```diff
@@ -8,48 +8,56 @@ export default class SeatPicker extends Component {
...
+    });
+  }
+
   saveHandler(){
-    let { boat } = this.props;
-    let { person, seatName } = this.props.activeSeat;
+    let { boat, seatName } = this.props;
+    let person = this.state.person;
 
     this.props.saveSeat(person, seatName, boat);
```
Comment (k3ntako) [inline client/pages/Boat/SeatPicker.js]: probably should delete boat

File: client/pages/Boat/Seat.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+  render(){
+    let seatClass = [styles.seat];
+    let seatName = this.props.seatName;
+    if( seatName === 'D' || seatName === 'S' ){
+      seatClass.push(styles.special);
+    }
+    if( this.props.canDrop ){
+      seatClass.push(styles.droppable);
+    }
```
Comment (k3ntako) [inline client/pages/Boat/Seat.js]: I think it'd be helpful to have the seat change color when you hover over it while dragging.

We could do something like this:
Add this here:
```
if( this.props.isOver ){
  seatClass.push(styles.isOver);
}
```

Add this to `index.css`
```
.seat.droppable.isOver{
  border-color: var(--palette-red-500);
}
```

File: client/pages/Boat/Seat.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        { children }
+        <SeatPicker
+          {...this.props}
+          isPickerActive={this.state.picker}
+          handlePickerToggle={this.closePicker}
+        />
+      </>
+    } else {
+      children = <div className={styles.paddler}>
+        Not Assigned
```
Comment (k3ntako) [inline client/pages/Boat/Seat.js]: Right now if the user is not a coach it only return `Not Assigned`. It never returns the paddler's name.

**Comment by k3ntako** [comment]
Other thing is that it's hard to tell that you can drag, but that's probably an issue for later

File: client/pages/Boat/SeatPicker.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mienaikoe) [reply client/pages/Boat/SeatPicker.js]: good eye, thanks.

File: client/pages/Boat/BoatPanel.js [hunk 4]
```diff
@@ -1,13 +1,12 @@
 import React, {Component} from 'react';
+
 import styles from './index.css';
-import Input from 'react-toolbox/lib/input/Input';
 import { getBoatsAssigned } from './getBoatsAssigned';
+import Paddler from './Paddler';
```
[the remark this answers is not in the document]
Comment (mienaikoe) [reply client/pages/Boat/BoatPanel.js]: true

File: client/pages/Boat/BoatPanel.js [hunk 5]
```diff
@@ -69,62 +70,39 @@ export default class Seats extends Component {
...
-      activeSeat={activeSeat}
-      isPickerActive={isPickerActive}
-      boatsAssigned={boatsAssigned}
-      />
+    this.state.boat.seat(person, seatName).then(followup).then( () => {
+      this.forceUpdate();
+    })
   }
 
   renderSeats(seatName, boatsAssigned){
```
[the remark this answers is not in the document]
Comment (mienaikoe) [reply client/pages/Boat/BoatPanel.js]: yeh I was thinking that but glad you said it too :p

File: client/pages/Boat/Paddler.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    connectDragSource: connect.dragSource(),
+    isDragging: monitor.isDragging()
+  };
+}
+
+class Paddler extends Component {
+
+  render(){
+    let className = [styles.paddler];
+    if( this.props.dragging ){
```
[the remark this answers is not in the document]
Comment (mienaikoe) [reply client/pages/Boat/Paddler.js]: ah yea. Missed that.

the library that lets us import css and use styles.className is the babel css loader. check out `css-loader` in npm with the modules option true.

[the remark this answers is not in the document]
Comment (mienaikoe) [reply client/pages/Boat/Paddler.js]: fixed

File: client/pages/Boat/index.css [hunk 7]
```diff
@@ -30,41 +30,43 @@
...
+  border-color: var(--palette-blue-grey-500);
+}
 
-.coachModeInput{
-  font-size: 1.4rem;
+.paddler{
+  padding: 1rem 1.5rem;
+}
+.paddler.dragging{
+  opacity: 0.5;
```
[the remark this answers is not in the document]
Comment (mienaikoe) [reply client/pages/Boat/index.css]: fixed

File: client/pages/Boat/Seat.js (hunk 2 above)
Comment (mienaikoe) [reply client/pages/Boat/Seat.js]: Idk about red, but I'll make it the pink we have in some cases

File: client/pages/Boat/Seat.js (hunk 3 above)
Comment (mienaikoe) [reply client/pages/Boat/Seat.js]: Oh jeez! Thanks for catching this!

**Comment by mienaikoe** [comment]
Yea, it is hard to tell, but I imagine once coach tries it once, he'll get it.

**Pull request merged**